Mathlib/Algebra/Lie/Nilpotent.lean

```diff
@@ -716,6 +718,24 @@ theorem isNilpotent_of_le (M₁ M₂ : LieSubmodule R L M) (h₁ : M₁ ≤ M₂
   have hfg : ∀ x m, ⁅f x, g m⁆ = g ⁅x, m⁆ := by aesop
   exact (Submodule.inclusion_injective h₁).lieModuleIsNilpotent hfg
 
+/-- The max nilpotent submodule is the `sSup` of all nilpotent submodules. -/
+def maxNilpotentSubmodule :=
+  sSup { N : LieSubmodule R L M | IsNilpotent L N }
+
+instance maxNilpotentSubmoduleIsNilpotent [IsNoetherian R M] :
+    IsNilpotent L (maxNilpotentSubmodule R L M) := by
+  have hwf := LieSubmodule.wellFoundedGT_of_noetherian R L M
+  rw [← CompleteLattice.isSupClosedCompact_iff_wellFoundedGT] at hwf
```

Comment:
Since `LieSubmodule.wellFoundedGT_of_noetherian` is an instance, it is supposed to be picked up automatically. Usually (though not always) something is not quite right when we need to address an instance by name. In this case we can golf as follows:
```suggestion
  have hwf := CompleteLattice.WellFoundedGT.isSupClosedCompact (LieSubmodule R L M) inferInstance
```
but we still shouldn't need that final `inferInstance`: the final argument to `CompleteLattice.WellFoundedGT.isSupClosedCompact` should be a typeclass argument rather than an explicit argument.

The reason for these minor issues is that it is only relatively recently that we promoted the concept of a well founded order to a typeclass (as `WellFoundedGT`).

In time, we'll smooth out these wrinkles.